treemacs.el

```diff
@@ -605,6 +605,58 @@ Delete all elements whose car is ‘eq’ to KEY from ALIST."
   (beginning-of-line)
   (forward-char))
 
+(defun treemacs--path-in-dir-p (path dir)
+  ;;"Is path in the directory dir?"
+  (let (case-fold-search)
+    (if path
+        (string-match-p (concat "^" (regexp-quote (concat dir "/"))) path)
+     )
+    )
+  )
+
+(defun treemacs--kill-buffers-after-deletion (path is-dir)
+  "Clean up after a deleted file or directory. When path is a file,
+just kill the buffer visiting this file. When path is a folder,
+go through the buffer list and kill all buffers if path is a
+prefix."
+
+  (if (not is-dir)
+      (let ((buf (get-file-buffer path)))
+        (and buf
+             (y-or-n-p (format "Kill buffer of %s, too? "
+                               (file-name-nondirectory path)))
+             (kill-buffer buf)))
+
+    ;; Prompt for each buffer visiting a file in directory
+    (dolist (buf (buffer-list))
+      (with-current-buffer buf
```

Comment:
Yes, either way works. With two aequivalent solutions I'd just go with the one introducing fewer new code blocks and indentation.